### PR TITLE
Add usage example to README.md

### PR DESCRIPTION
**Pull request by maxmynter**
Added a usage example to `README.md`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,77 @@ poetry add git+https://github.com/aai-institute/nnbench.git
...
+data = load_iris()
+X, y = data.data, data.target
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3)
+
+model = RandomForestClassifier()
+model.fit(X_train, y_train)
+```
+
+To benchmark your model, you encapsulate the benchmark code into a function and assign the `@benchmark` decorator. This let's our benchmark class collect and later execute the function.
```
Comment (nicholasjng) [inline README.md:46]:
```suggestion
To benchmark your model, you encapsulate the benchmark code into a function and apply the `@benchmark` decorator. This marks the function for collection to our benchmark runner later.
```

File: README.md [hunk 2]
```diff
@@ -24,6 +24,77 @@ poetry add git+https://github.com/aai-institute/nnbench.git
...
+from training import model, X_test, y_test
+
+
+PATH_TO_BENCHMARK_DIR = pathlib.Path('.').resolve()
+
+
+class ExampleRunner(runner.AbstractBenchmarkRunner):
+    # Here you could integrate your custom benchmark handling logic.
+    # We keep it empty in this example
+    pass
```
Comment (nicholasjng) [inline README.md:76]: You did the right thing here, but we shouldn't advertise this in the readme. Therefore, we should probably rename the abstract class to just `BenchmarkRunner`.

File: README.md [hunk 3]
```diff
@@ -24,6 +24,77 @@ poetry add git+https://github.com/aai-institute/nnbench.git
...
+
+
+r = ExampleRunner()
+r.collect()
+y_pred = model.predict(X_test)
+
+result = r.run(PATH_TO_BENCHMARK_DIR, tags=("accuracy",),  params={"model": model, "y_pred": y_pred, "y_test": y_test})
```
Comment (nicholasjng) [inline README.md:83]: This should probably work with "benchmark.py"? Then you can save the `PATH_TO_...` definition, and drop the tags. (Less is more!)

File: README.md [hunk 4]
```diff
@@ -24,6 +24,77 @@ poetry add git+https://github.com/aai-institute/nnbench.git
...
+
+r.report(to='console', result=result)
+```
+The output then looks like this
```
Comment (nicholasjng) [inline README.md:87]:
```suggestion
The resulting output might look like this:
```

**Review by nicholasjng — COMMENTED** [review]
Great job!

File: README.md (hunk 3 above)
Comment (maxmynter) [reply README.md:83]: it does with `"./benchmark.py"`.

File: README.md [hunk 5]
```diff
@@ -24,6 +24,69 @@ poetry add git+https://github.com/aai-institute/nnbench.git
 
 A PyPI package release is planned for the future.
 
+## Usage Example
+
+An example repository to benchmark a ML model contains three files. The `training.py` contains the code to train the model, the `benchmark.py` contains the benchmark functions, and the `run_benchmark.py` the benchmark execution logic.
```
Comment (nicholasjng) [inline README.md:29]: For this example it's fine, but I'm not sure if we want to advertise that file structure regardless.

It makes sense if we think about benchmarks in a MLOps pipeline way. Question is, do we still include it in the introductory example to only showcase the basic application?

I think this example should serve as the Quickstart document, and be a single-file monolith to convey the simplicity of the package.

Then, if the user is interested, we link to user guides (to be written) on file structure/organization, tagging, subclassing runners,reporters,etcpp. What do you think?

**Review by nicholasjng — APPROVED** [review]
Nice! Check this in and then let's talk about restructuring later. 

Please rebase as two commits , one for the example and one for the runner rename.

To do that, you can either use `git rebase -i` or `git reset main` and then craft the two commits manually by adding the respective file set (changes should be orthogonal in this case).

**Review by nicholasjng — APPROVED** [review]
